screenlayout/widget.py

```diff
@@ -199,47 +229,57 @@ def _draw(self, xrandr, cr):
         cfg = xrandr.configuration
         state = xrandr.state
 
-        cr.set_source_rgb(0.25,0.25,0.25)
-        cr.rectangle(0,0,*state.virtual.max)
+        cr.set_source_rgb(0.25, 0.25, 0.25)
+        cr.rectangle(0, 0, *state.virtual.max)
         cr.fill()
 
-        cr.set_source_rgb(0.5,0.5,0.5)
-        cr.rectangle(0,0,*cfg.virtual)
+        cr.set_source_rgb(0.5, 0.5, 0.5)
+        cr.rectangle(0, 0, *cfg.virtual)
         cr.fill()
 
         for on in self.sequence:
             o = cfg.outputs[on]
-            if not o.active: continue
+            if not o.active:
+                continue
 
-            rect = (o.tentative_position if hasattr(o, 'tentative_position') else o.position) + tuple(o.size)
+            rect = (o.tentative_position if hasattr(
+                o, 'tentative_position') else o.position) + tuple(o.size)
             center = rect[0]+rect[2]/2, rect[1]+rect[3]/2
 
             # paint rectangle
-            cr.set_source_rgba(1,1,1,0.7)
+            cr.set_source_rgba(1, 1, 1, 0.7)
             cr.rectangle(*rect)
             cr.fill()
-            cr.set_source_rgb(0,0,0)
+            cr.set_source_rgb(0, 0, 0)
             cr.rectangle(*rect)
             cr.stroke()
 
             # set up for text
             cr.save()
             textwidth = rect[3 if o.rotation.is_odd else 2]
             widthperchar = textwidth/len(on)
-            textheight = int(widthperchar * 0.8) # i think this looks nice and won't overflow even for wide fonts
+            # i think this looks nice and won't overflow even for wide fonts
+            textheight = int(widthperchar * 0.8)
 
-            newdescr = pango.FontDescription("sans")
-            newdescr.set_size(textheight * pango.SCALE)
+            newdescr = Pango.FontDescription("sans")
+            newdescr.set_size(textheight * Pango.SCALE)
 
             # create text
-            layout = cr.create_layout()
+            layout = PangoCairo.create_layout(cr)
             layout.set_font_description(newdescr)
             if o.primary:
-                attrs = pango.AttrList()
-                attrs.insert(pango.AttrUnderline(pango.UNDERLINE_SINGLE, end_index=-1))
-                layout.set_attributes(attrs)
-
-            layout.set_text(on)
+                underline_attrs = Pango.parse_markup("<u>test</u>", -1, "0")[1]
```

Comment:
this is the only part of my port which i'm not proud of, i spend more than 10 minutes but didn't Underline Attribute to work in other way than this